optuna/study.py

```diff
@@ -578,6 +578,41 @@ def trials(self):
 
         return self.storage.get_all_trials(self.study_id)
 
+    @property
+    def product_search_space(self):
+        # type: () -> Dict[str, BaseDistribution]
+        """Return the product search space of the :class:`~optuna.study.InTrialStudy`.
+
+        "product search space" contains the product of parameter distributions that have been
```

Comment:
IMO, the expression 'product of parameter distributions' seems to be confusing with [product distribution](https://en.wikipedia.org/wiki/Product_distribution). Can we rephrase this as something like 'product set of parameter distributions'?